cart/src/main/java/com/yas/cart/service/CartService.java

```diff
@@ -142,4 +144,12 @@ public void removeCartItemByProductId(Long productId) {
 
         cartItemRepository.deleteByCartIdAndProductId(currentCart.id(), productId);
     }
+
+    public Flux<Integer> countNumberItemInCart(String customerId) {
+        return Flux.interval(Duration.ofSeconds(1)).map((i) -> {
+            Cart cart = cartRepository.findByCustomerId(customerId)
+                    .stream().reduce((first, second) -> second).orElse(null);
+            return cartItemRepository.countItemInCart(cart.getId());
+        });
+    }
```

Comment:
Add a null check for the cart variable and return an appropriate value, such as an empty Mono or Flux stream, or a default value like 0. This will prevent a NullPointerException from being thrown.